packages/grid/x-data-grid/src/models/api/gridPreferencesPanelApi.ts

```diff
@@ -7,8 +7,18 @@ export interface GridPreferencesPanelApi {
   /**
    * Displays the preferences panel. The `newValue` argument controls the content of the panel.
    * @param {GridPreferencePanelsValue} newValue The panel to open. Use `"filters"` or `"columns"`.
+
+   * @param {Object?} ids Object containing button and panel ids
+   * @param {string?} ids.panelId The unique panel id
+   * @param {string?} ids.buttonId The unique button id
    */
-  showPreferences: (newValue: GridPreferencePanelsValue) => void;
+  showPreferences: (
+    newValue: GridPreferencePanelsValue,
+    ids?: {
+      panelId?: string;
+      buttonId?: string;
+    },
+  ) => void;
```

Comment:
What do you think of this signature? It's more generic and we don't need to always pass the panel ID, since there's always only one panel.

```suggestion
  showPreferences: (
    newValue: GridPreferencePanelsValue,
    labelId?: string,
  ) => void;
```

To replace the panel ID argument you need to initialize the state with an ID. Here's an example of how it can be done:

```diff
diff --git a/packages/grid/x-data-grid/src/DataGrid/useDataGridComponent.tsx b/packages/grid/x-data-grid/src/DataGrid/useDataGridComponent.tsx
index 3eed1efbf..7978b08be 100644
--- a/packages/grid/x-data-grid/src/DataGrid/useDataGridComponent.tsx
+++ b/packages/grid/x-data-grid/src/DataGrid/useDataGridComponent.tsx
@@ -53,6 +53,7 @@ import { useGridColumnGroupingPreProcessors } from '../hooks/features/columnGrou
 
 export const useDataGridComponent = (props: DataGridProcessedProps) => {
   const apiRef = useGridInitialization<GridApiCommunity>(undefined, props);
+  const panelId = useId();
 
   /**
    * Register all pre-processors called during state initialization here.
@@ -77,7 +78,7 @@ export const useDataGridComponent = (props: DataGridProcessedProps) => {
   );
   useGridInitializeState(focusStateInitializer, apiRef, props);
   useGridInitializeState(sortingStateInitializer, apiRef, props);
-  useGridInitializeState(preferencePanelStateInitializer, apiRef, props);
+  useGridInitializeState(preferencePanelStateInitializer, apiRef, { ...props, panelId });
   useGridInitializeState(filterStateInitializer, apiRef, props);
   useGridInitializeState(densityStateInitializer, apiRef, props);
   useGridInitializeState(paginationStateInitializer, apiRef, props);
diff --git a/packages/grid/x-data-grid/src/hooks/features/preferencesPanel/useGridPreferencesPanel.ts b/packages/grid/x-data-grid/src/hooks/features/preferencesPanel/useGridPreferencesPanel.ts
index 473a2634d..cb9fc8f31 100644
--- a/packages/grid/x-data-grid/src/hooks/features/preferencesPanel/useGridPreferencesPanel.ts
+++ b/packages/grid/x-data-grid/src/hooks/features/preferencesPanel/useGridPreferencesPanel.ts
@@ -9,10 +9,10 @@ import { GridPreferencesPanelApi } from '../../../models/api/gridPreferencesPane
 import { GridStateInitializer } from '../../utils/useGridInitializeState';
 
 export const preferencePanelStateInitializer: GridStateInitializer<
-  Pick<DataGridProcessedProps, 'initialState'>
+  Pick<DataGridProcessedProps, 'initialState'> & { panelId: string }
 > = (state, props) => ({
   ...state,
-  preferencePanel: props.initialState?.preferencePanel ?? { open: false },
+  preferencePanel: props.initialState?.preferencePanel ?? { open: false, panelId: props.panelId },
 });
 
 /**
```

You may need to adjust the `GridStateInitializer` interface to accept other props, or add another argument to the state initializer type.

Reply:
I think you are right. I just checked again and it is the same panel instance for both column and filter buttons. so it make sense to keep just pass the label_ID

Reply:
Not applied in #8862. `useId` value during the first render is `undefined`, that patch isn't easily applicable.